packages/x-data-grid/src/hooks/features/rows/gridRowsUtils.ts

```diff
@@ -393,3 +403,35 @@ export function getMinimalContentHeight(apiRef: React.MutableRefObject<GridApiCo
   const dimensions = gridDimensionsSelector(apiRef.current.state);
   return `var(--DataGrid-overlayHeight, ${2 * dimensions.rowHeight}px)`;
 }
+
+export function computeRowsUpdates(
```

Comment:
For my information, why is this needed now?

Reply:
This code is moved from the previous `updateRows` method to be reused in `updateServerRows`.
Context: https://github.com/mui/mui-x/pull/12317#discussion_r1633145117